python/pyspark/pandas/frame.py

```diff
@@ -1310,6 +1310,137 @@ def corr(self, method: str = "pearson") -> "DataFrame":
+            combined = self
+            this = self
+            that = other
+        else:
+            combined = combine_frames(self, other, how="inner")
+            this = combined["this"]
+            that = combined["that"]
+
+        this_numeric_column_labels: List[Label] = []
+        for column_label in this._internal.column_labels:
+            if isinstance(this._internal.spark_type_for(column_label), (NumericType, BooleanType)):
+                this_numeric_column_labels.append(column_label)
+
+        that_numeric_column_labels: List[Label] = []
+        for column_label in that._internal.column_labels:
+            if isinstance(that._internal.spark_type_for(column_label), (NumericType, BooleanType)):
+                that_numeric_column_labels.append(column_label)
+
+        intersect_numeric_column_labels: List[Label] = []
+        diff_numeric_column_labels: List[Label] = []
+        corr_scols = []
+        if right_is_series:
+            intersect_numeric_column_labels = this_numeric_column_labels
+            that_scol = that._internal.spark_column_for(that_numeric_column_labels[0])
+            for numeric_column_label in intersect_numeric_column_labels:
+                this_scol = this._internal.spark_column_for(numeric_column_label)
+                corr_scols.append(F.corr(this_scol.cast("double"), that_scol.cast("double")))
+        else:
+            for numeric_column_label in this_numeric_column_labels:
+                if numeric_column_label in that_numeric_column_labels:
+                    intersect_numeric_column_labels.append(numeric_column_label)
+                else:
+                    diff_numeric_column_labels.append(numeric_column_label)
+            for numeric_column_label in that_numeric_column_labels:
+                if numeric_column_label not in this_numeric_column_labels:
+                    diff_numeric_column_labels.append(numeric_column_label)
+            for numeric_column_label in intersect_numeric_column_labels:
+                this_scol = this._internal.spark_column_for(numeric_column_label)
+                that_scol = that._internal.spark_column_for(numeric_column_label)
+                corr_scols.append(F.corr(this_scol.cast("double"), that_scol.cast("double")))
+
+        ret_column_labels = []
+        ret_values = []
+        if len(intersect_numeric_column_labels) > 0:
+            ret_column_labels = intersect_numeric_column_labels
+            pcorr = combined._internal.spark_frame.select(*corr_scols).toPandas()
+            ret_values = [pcorr.iloc[0, i] for i in range(len(intersect_numeric_column_labels))]
+
+        if not drop and len(diff_numeric_column_labels) > 0:
+            for numeric_column_label in diff_numeric_column_labels:
+                ret_column_labels.append(numeric_column_label)
+                ret_values.append(np.nan)
+
+        idx = pd.Index([label[0] for label in ret_column_labels])
+        return ps.Series(ret_values, idx)
```

Comment:
Hm .. this one I don't have a strong opinion but wonder if it's better to call `transpose()` (like `DataFrame.mad`) so it does not eagerly executes the job with `toPandas` above. I know that we already have such places but I am thinking that triggering the job execution within an API is something we should avoid if possible.

Reply:
let me have a try